### PR TITLE
Finished setting up the react native app. It works now.

### PR DESCRIPTION
**Pull request by Hasintha-Jayasinghe**

### CODE REVIEW

File: kofta/src/vscode-webview/App.tsx [hunk 1]
```diff
@@ -3,9 +3,9 @@ import queryString from "query-string";
 import React, { useEffect, useLayoutEffect, useState } from "react";
 import { QueryClientProvider } from "react-query";
 import { tw } from "twind";
-import { createWebSocket } from "../createWebsocket";
-import { useSocketStatus } from "../webrtc/stores/useSocketStatus";
-import { useWsHandlerStore } from "../webrtc/stores/useWsHandlerStore";
+import { createWebSocket } from "feta/createWebsocket";
+import { useSocketStatus } from "feta/webrtc/stores/useSocketStatus";
+import { useWsHandlerStore } from "feta/webrtc/stores/useWsHandlerStore";
```
Comment (benawad) [inline kofta/src/vscode-webview/App.tsx]: I want imports to be `@dogehouse/feta`

File: pilaf/package.json [hunk 2]
```diff
@@ -7,9 +7,13 @@
     "ios": "react-native run-ios",
     "start": "react-native start",
     "test": "jest",
-    "lint": "eslint ."
+    "lint": "eslint .",
+    "postinstall": "react-native-yarn-workspaces-v2"
```
Comment (benawad) [inline pilaf/package.json:11]: does this use yarn v2?

**Comment by overlisted** [comment]
Also add the component's description to the table in our main README?

File: feta/constants.ts [hunk 3]
```diff
@@ -4,4 +4,4 @@ export const apiBaseUrl =
   process.env.REACT_APP_API_BASE_URL ||
   (__prod__
     ? "https://api.dogehouse.tv"
-    : "https://doge-staging.stripcode.dev/help");
+    : "https://doge-staging.stripcode.dev");
```
Comment (benawad) [inline feta/constants.ts]: this file should not be changed

File: feta/createWebsocket.ts [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-import { WsParam } from "./vscode-webview/types";
+import { useTokenStore } from "kofta/src/vscode-webview/utils/useTokenStore";
+import { showErrorToast } from "kofta/src/vscode-webview/utils/showErrorToast";
+import { apiBaseUrl } from "./constants";
+import { useSocketStatus } from "kofta/src/webrtc/stores/useSocketStatus";
+import { useWsHandlerStore } from "kofta/src/webrtc/stores/useWsHandlerStore";
+import { useVoiceStore } from "kofta/src/webrtc/stores/useVoiceStore";
+import { useMuteStore } from "kofta/src/webrtc/stores/useMuteStore";
+import { uuidv4 } from "kofta/src/webrtc/utils/uuidv4";
+import { WsParam } from "kofta/src/vscode-webview/types";
```
Comment (benawad) [inline feta/createWebsocket.ts]: `feta` should not import anything from `kofta`, you should move all imports over

**Comment by overlisted** [comment]
why does it do touch kofta at all

**Comment by benawad** [comment]
because it needs to move functions that both the website and react native app will need to commons folder

**Comment by overlisted** [comment]
oh and feta is that component?

**Comment by benawad** [comment]
yeah feta is the shared folder

**Comment by overlisted** [comment]
why is he doing this already for 4 days

File: feta/webrtc/utils/sendVoice.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
-import { useMicIdStore } from "../../vscode-webview/shared-stores";
-import { useMicPermErrorStore } from "../stores/useMicPermErrorStore";
-import { useProducerStore } from "../stores/useProducerStore";
-import { useVoiceStore } from "../stores/useVoiceStore";
+import { useMicIdStore } from "../../../kofta/src/vscode-webview/shared-stores";
```
Comment (benawad) [inline feta/webrtc/utils/sendVoice.ts]: this import needs to be feta

File: kofta/src/vscode-webview/components/CreateRoomModal.tsx [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import React, { useState } from "react";
-import { wsend } from "../../createWebsocket";
+import { tw } from "twind";
```
Comment (benawad) [inline kofta/src/vscode-webview/components/CreateRoomModal.tsx]: twind is gone, remove this import

File: kofta/src/vscode-webview/components/EditProfileModal.tsx [hunk 7]
```diff
@@ -3,11 +3,12 @@ import { useAtom } from "jotai";
 import React from "react";
 import { useMutation } from "react-query";
 import { object, pattern, size, string } from "superstruct";
-import { wsFetch } from "../../createWebsocket";
+import { tw } from "twind";
```
Comment (benawad) [inline kofta/src/vscode-webview/components/EditProfileModal.tsx]: ^^

File: kofta/src/vscode-webview/components/MicPermissionBanner.tsx [hunk 8]
```diff
@@ -1,7 +1,8 @@
 import React, { useState } from "react";
-import { useMicPermErrorStore } from "../../webrtc/stores/useMicPermErrorStore";
-import { sendVoice } from "../../webrtc/utils/sendVoice";
-import { isIOS } from "../utils/isIOS";
+import { tw } from "twind";
```
Comment (benawad) [inline kofta/src/vscode-webview/components/MicPermissionBanner.tsx]: ^^

File: kofta/src/vscode-webview/components/ProfileModal.tsx [hunk 9]
```diff
@@ -1,9 +1,10 @@
 import React from "react";
 import ReactModal from "react-modal";
-import { wsend } from "../../createWebsocket";
+import { tw } from "twind";
```
Comment (benawad) [inline kofta/src/vscode-webview/components/ProfileModal.tsx]: ^^

File: kofta/src/vscode-webview/components/RoomCard.tsx [hunk 10]
```diff
@@ -54,9 +54,21 @@ export const RoomCard: React.FC<RoomProps> = ({
         <div className={`flex text-white`}>
           <div
             style={{
+<<<<<<< HEAD
```
Comment (benawad) [inline kofta/src/vscode-webview/components/RoomCard.tsx]: see staging for this merge conflict

File: kofta/src/vscode-webview/components/RoomSettingsModal.tsx [hunk 11]
```diff
@@ -1,9 +1,16 @@
 import { useAtom } from "jotai";
 import React from "react";
 import { X } from "react-feather";
+<<<<<<< HEAD
 import { wsend } from "../../createWebsocket";
 import { renameRoomAndMakePrivate } from "../../webrtc/utils/renameRoomAndMakePrivate";
 import { renameRoomAndMakePublic } from "../../webrtc/utils/renameRoomAndMakePublic";
+=======
+import { tw } from "twind";
```
Comment (benawad) [inline kofta/src/vscode-webview/components/RoomSettingsModal.tsx]: to resolve this conflict, kill twind

File: kofta/src/vscode-webview/components/UserProfile.tsx [hunk 12]
```diff
@@ -1,14 +1,16 @@
 import React, { useEffect, useState } from "react";
 import { useAtom } from "jotai";
 import { useHistory } from "react-router-dom";
-import { wsend } from "../../createWebsocket";
+import { tw } from "twind";
```
Comment (benawad) [inline kofta/src/vscode-webview/components/UserProfile.tsx]: ^^

File: kofta/src/vscode-webview/modules/room-chat/RoomChatInput.tsx [hunk 13]
```diff
@@ -104,60 +105,14 @@ export const RoomChatInput: React.FC<ChatInputProps> = ({}) => {
...
-          className={`text-simple-gray-9c bg-simple-gray-59 px-4 py-3 rounded text-lg focus:outline-none pr-12`}
-          onKeyDown={navigateThroughQueriedUsers}
-          onFocus={() => {
-            setisEmoji(false);
-            position = 0;
-          }}
-          id="room-chat-input"
-        />
-      </div>
+      <input
```
Comment (benawad) [inline kofta/src/vscode-webview/modules/room-chat/RoomChatInput.tsx:107]: don't change this code, grab it from staging

**Comment by overlisted** [comment]
it looks like ben is writing the component for you

File: feta/webrtc/utils/invitationToRoom.ts [hunk 14]
```diff
@@ -1,4 +1,4 @@
-import { wsend } from "../../createWebsocket";
+import { wsend } from "../../../../feta/createWebsocket";
```
Comment (benawad) [inline feta/webrtc/utils/invitationToRoom.ts]: this should not `../feta`

**Pull request merged**